__tests__/integration.js

```diff
@@ -233,16 +233,12 @@ describe('--registry option', () => {
     expect(lockfile[2]).toContain(registry);
   });
 
-  test('--registry option with non-exiting registry and show an error', async () => {
+  test('--registry option with nonexistent registry and show an error', async () => {
     const cwd = await makeTemp();
-    const registry = 'https://example-registry-doesnt-exist.com';
+    const registry = 'https://example-registry-doesnt-exist.invalid'; // RFC 6761
 
-    try {
-      await runYarn(['add', 'is-array', '--registry', registry], {cwd});
-    } catch (err) {
-      const stdoutOutput = err.message;
-      expect(stdoutOutput.toString()).toMatch(/getaddrinfo ENOTFOUND example-registry-doesnt-exist\.com/g);
-    }
+    const yarnAdd = runYarn(['add', 'is-array', '--registry', registry, '--ignore-scripts'], {cwd});
+    await expect(yarnAdd).rejects.toThrow('getaddrinfo ENOTFOUND example-registry-doesnt-exist.invalid');
```

Comment:
Note some DNSs time out on `.invalid` domains throwing `EAI_AGAIN`.

Reply:
Thanks @ylemkimon - I wasn't aware of that. Updated the test to expect either `ENOTFOUND` or `EAI_AGAIN` from `getaddrinfo`. 

Let me know what you think.